File: CountVowelsAndConsonants.py
# CountVowelsAndConsonants
# - Given a string s, count vowels and consonants.
# - Return a tuple (vowels_count, consonants_count).
# - Time: O(n), Space: O(1)

class Solution:
    def countVowelsAndConsonants(self, s: str) -> tuple[int, int]:
        vowels = "aeiouAEIOU"
        vCount, cCount = 0, 0
        for ch in s:
            if ch.isalpha():
                if ch in vowels:
                    vCount += 1
                else:
                    cCount += 1
        return (vCount, cCount)

File: CountVowelsAndConsonants.cs
// CountVowelsAndConsonants
// - Given a string s, count vowels and consonants.
// - Return a tuple (vowels_count, consonants_count).
// - Time: O(n), Space: O(1)

using System;

public class Solution {
    public (int, int) CountVowelsAndConsonants(string s) {
        string vowels = "aeiouAEIOU";
        int vCount = 0, cCount = 0;

        foreach (char ch in s) {
            if (char.IsLetter(ch)) {
                if (vowels.Contains(ch)) vCount++;
                else cCount++;
            }
        }

        return (vCount, cCount);
    }
}
